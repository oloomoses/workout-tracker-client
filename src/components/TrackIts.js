import React from 'react';
import { useSelector } from 'react-redux';
import { Card } from 'react-bootstrap';
import RPB from 'react-radial-progress-bar';
import Spinner from './Spinner';

const TrackIts = () => {
  const loading = useSelector((state) => state.trackIt.loading);
  const trackIts = useSelector((state) => state.trackIt.trackIts);
  if (loading) {
    return <Spinner />;
  }

  const toPercentage = (val, base) => {
    const toInt = parseInt(val, 10);
    const toP = (toInt / base) * 100;
    return toP.toString();
  };

  return (
    <div className="tracks-wrapper">
      { trackIts.map((track) => (
        <Card key={track.id} className="track-card-layout mt-3">
          <div className="d-flex justify-content-between align-items-center mx-3 my-2">
            <div className="d-flex align-items-center">

              <div className="mr-3">
                <RPB
                  size={70}
                  percentageCompleted={toPercentage(track.calories, 5000)}
                  fillColor="green"
                  backgroundColor="#d6dadc"
                />
              </div>

              <div>
                <div className="track-date">{ new Date(track.created_at).toDateString() }</div>
                <div className="track-calories">
                  Calories:
                  {' '}
                  {toPercentage(track.calories, 5000)}
                  .0%
                </div>
              </div>

            </div>
            <div>
              <div className="track-mins">
                {track.minutes_recorded}
                mins
              </div>
            </div>
          </div>
        </Card>
      ))}
    </div>

  );
};

export default TrackIts;
