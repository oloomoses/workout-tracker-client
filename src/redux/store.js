import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import userRegistrationReducer from './user/userRegisterReducer';

const store = createStore(userRegistrationReducer, applyMiddleware(thunk));

export default store;
