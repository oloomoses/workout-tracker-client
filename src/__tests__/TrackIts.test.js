import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import TrackIts from '../components/TrackIts';
import store from '../redux/store';

describe('TrackIts component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<Provider store={store}><TrackIts /></Provider>).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
