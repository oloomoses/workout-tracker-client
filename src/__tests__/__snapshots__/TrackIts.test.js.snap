// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackIts component Matches the snapshot 1`] = `[Function]`;
